### ISSUE TITLE
Problems finding openssl headers on MacOS 10.14.6 and how to fix it

### ISSUE THREAD

**Issue by jahzielv** [MasterQ32/kristall#62]
I wanted to build Kristall on my MacBook Pro 2017 running MacOS 10.14.6, but `make build/kristall` errored out:
![image](https://user-images.githubusercontent.com/19838370/92330865-3ec6a380-f040-11ea-8861-a204244d2df2.png)

Openssl was already installed on my machine, but I ran `brew reinstall openssl@1.1` just in case; no cigar.

Next, I tried exporting the include path like so
```bash
 export LDFLAGS="-L/usr/local/opt/openssl/lib"
 export CFLAGS="-I/usr/local/opt/openssl/include/"
 export CXXFLAGS="-I/usr/local/opt/openssl/include/"
```
but this didn't work either. Next, I tried modifying `kristall-0.3/build/Makefile` to add the openssl include path directly to the `INCPATH` variable, but for some reason the file would get edited back to its original state when running `make build/kristall`??

So finally, after reading the root Makefile and mostly figuring out what the building process was, I added the openssl include path to the `INCPATH` variable in the `build` Makefile, and ran `make` within the `build` directory. This finally worked!! Everything built smoothly, and I was able to run `Kristall.app` and browse Gemini sites.

Has anyone else had the same problem? If so, I would be down to edit the documentation for building on MacOS. Thanks for working on this project; it's a really nice browser!

**Comment by MasterQ32** [comment]
Thanks for the writeup!

> but for some reason the file would get edited back to its original state when running make build/kristall??

This is because the `Makefile` in `build/` is generated by `kristall.pro`

I would appreciate some more tips about building Kristall, i'm happily accepting PRs!

**Comment by jahzielv** [comment]
> This is because the Makefile in build/ is generated by kristall.pro

Ahh ok! I took a look in `kristall.pro`, but I'm not familiar with qmake so I wasn't sure what all was going on in there. Thanks for explaining! Should I open a PR with a note about how to fix these errors?

**Comment by MasterQ32** [comment]
> Should I open a PR with a note in BUILDING.md about how to fix these errors?
Yes, please! You could even create a mac-specific patch to `kristall.pro` fixing the paths!

**Comment by jagger27** [comment]
I believe when I added the Qt hack into the current Makefile I had openssl installed via Homebrew.
